compiler-collection.js

```diff
@@ -0,0 +1,49 @@
+module.exports = (expireAfterSeconds) => {
+  const expireMs = expireAfterSeconds * 1000;
+  const compilers = new Map();
+
+  if (expireAfterSeconds) {
+    // Check for expired things often enough to expire things within
+    // 20% of their true expiration time.
+    setInterval(removeExpiredCompilers, expireAfterSeconds / 5);
```

Comment:
It occurs to me that a simpler way to do this might be to register a per-compiler timeout whenever we would `updateLastAccessed` for `expireMs` and cancel any existing timeout at the same time. I think it's basically as easy as storing the last accessed time, and then we expire things exactly when we expect to. Does that make sense?

Reply:
Not a bad idea. I think that's probably a bit more intuitive and probably less code. Doing it!